examples/fibonacci.rs

```diff
@@ -162,4 +167,23 @@ fn main() {
             println!("{}", failure);
         }
     }
+
+    // hyperplonk boilerplate
+    use hyperplonk_benchmark::proof_system::{bench_plonkish_backend, System};
+    use plonkish_backend::{
+        backend,
+        halo2_curves::bn256::{Bn256, Fr},
+        pcs::{multilinear, univariate},
+    };
+    // get Chiquito ir
+    let (circuit, assignment_generator) = fibo_circuit::<Fr>();
+    // get assignments
+    let assignments = assignment_generator.unwrap().generate(());
+    // get hyperplonk circuit
+    let mut hyperplonk_circuit = ChiquitoHyperPlonkCircuit::new(4, circuit);
+    hyperplonk_circuit.set_assignment(assignments);
```

Comment:
@qwang98 so what happens if you don't set the assigments, the circuit gets compiled?

Reply:
You need to be able to compile the circuit to hyperplonk, and then on the compiled circuit to be able to generate proof with the assignments. The assignments are needed in the proof generation phase, not for the compilation.

As you can see in the halo2 example.

Reply:
> @qwang98 so what happens if you don't set the assigments, the circuit gets compiled?

Yes. The circuit gets compiled even without the assignments. For example, constraints are converted to HyperPlonk format without needing assignments.

Reply:
> You need to be able to compile the circuit to hyperplonk, and then on the compiled circuit to be able to generate proof with the assignments. The assignments are needed in the proof generation phase, not for the compilation.
> 
> As you can see in the halo2 example.

This is what we have, but I believe your point is that only the proof generation function should take the assignment, no the compilation function, so I modified the public APIs to reflect this change. Let me know if I'm understanding correctly.